scripting/include/srccoop/classdef.inc

```diff
@@ -848,6 +1041,163 @@ methodmap CSceneEntity < CBaseEntity
 	}
 }
 
+methodmap CPointEntity < CBaseEntity
+{
+	public CPointEntity(const int iEntIndex = -1)
+	{
+		return view_as<CPointEntity>(CBaseEntity(iEntIndex));
+	}
+}
+
+methodmap CSoundEnt < CPointEntity
+{
+	public CSoundEnt(const int iEntIndex = -1)
+	{
+		return view_as<CSoundEnt>(CBaseEntity(iEntIndex));
+	}
+
+	// Gets the index of the first sound in the active sound list.
+	public void ActiveList(CSound pSound)
+	{
+		pSound.m_pSoundEnt = view_as<int>(this);
+		pSound.m_iSoundIndex = GetEntProp(this.GetEntIndex(), Prop_Data, "m_iActiveSound");
+	}
+}
+
+enum struct CSound
+{
+	int m_pSoundEnt;
```

Comment:
Is there a reason for not storing as CSoundEnt directly?

Reply:
No there currently isn't any reason. Changed implementation to directly use `CSoundEnt`.